daffodil-runtime1/src/main/scala/org/apache/daffodil/debugger/InteractiveDebugger.scala

```diff
@@ -1720,7 +1722,7 @@ class InteractiveDebugger(runner: InteractiveDebuggerRunner, eCompilers: Express
                         |Immediately abort all processing.""".stripMargin
       override lazy val short = "q"
       def act(args: Seq[String], state: ParseOrUnparseState, processor: Processor): DebugState.Type = {
-        sys.exit(1)
+        throw new DebuggerExitException()
```

Comment:
Nothing catches this exception. This should be caught in Main.scala so that we can exit with an appropriate exit code. A generic `ExitCode.Failure` seems reasonable.